src/hooks/useGridWidth.ts

```diff
@@ -1,23 +1,51 @@
-import { useRef, useState, useLayoutEffect } from 'react';
+import { useRef, useState, useLayoutEffect, useMemo } from 'react';
+
+// https://github.com/microsoft/TypeScript/issues/37861
+type ResizeObserverCallback = () => void;
+
+type ResizeObserver = new (callback: ResizeObserverCallback) => {
+  observe: (target: Element) => void;
+  disconnect: () => void;
+};
 
 export function useGridWidth<T>(width?: number): [React.RefObject<HTMLDivElement>, number] {
   const gridRef = useRef<HTMLDivElement>(null);
   const [gridWidth, setGridWidth] = useState(0);
+  const resizeObserver = useMemo(() => {
+    const { ResizeObserver } = window as (typeof window & { ResizeObserver?: ResizeObserver });
+    if (ResizeObserver === undefined) {
```

Comment:
Let's avoid setting up a resize observer if we won't end up using it.
```suggestion
    if (typeof width === 'number' || ResizeObserver === undefined) {
```